### DISCUSSION TITLE
Discriminated unions coming to C# (type unions)

### DISCUSSION THREAD

**Discussion by StefanBertels** [louthy/language-ext#1347] [General]
Seems that after a long discussion DU might arrive in C#:

https://github.com/dotnet/csharplang/blob/main/proposals/TypeUnions.md 

They want to introduce `Option` and `Result` (~Either/Fin). But: "no monadic behaviour" (for now). Hmm.

Could/Should lang-ext make use of those types union feature? Naming conflicts?

What do you think?

**Comment by louthy** [comment]
If they add Option to System then I'll remove it from lang-ext and then extend it to have the same interface.  

The biggest issue will be inability to use the traits. 

Although, I'm hoping that the new extension-types feature (which will come before sum-types) will enable ad-hoc traits. If that's the case then the transition should be quite smooth.
